selfservice/flow/login/handler.go

```diff
@@ -187,8 +188,17 @@ preLoginHook:
 		f.UI.Messages.Add(text.NewInfoLoginMFA())
 	}
 
-	var s Strategy
-	for _, s = range h.d.LoginStrategies(r.Context()) {
+	var strategyFilters []StrategyFilter
+	if rawOrg := r.URL.Query().Get("organization"); rawOrg != "" {
+		orgID, err := uuid.FromString(rawOrg)
+		if err != nil {
+			h.d.Logger().WithError(err).Warnf("ignoring invalid UUID %q in query parameter `organization`", rawOrg)
+		} else {
+			f.OrganizationID = uuid.NullUUID{UUID: orgID, Valid: true}
+			strategyFilters = []StrategyFilter{func(s Strategy) bool { return s.ID() == identity.CredentialsTypeOIDC }}
```

Comment:
Not sure if I understand why we need this filter. Could you explain?

Reply:
Because we want to filter out all other auth methods for the Org-SSO case.